### ISSUE TITLE
Cannot load old maps in ver 1.3.0.8 properly but can load new maps (1.3.1.0)

### ISSUE THREAD

**Issue by mistzzt** [nicatronTg/early-api#16]
**Here is what I got in debugging process**
In `TileEntity.cs` :
```
public static TileEntity Read(BinaryReader reader, bool networkSend = false)
		{
			TileEntity tileEntity = null;
			byte b = reader.ReadByte();
			switch (b)
			{
			case 0:
				tileEntity = new TETrainingDummy();
				break;
			case 1:
				tileEntity = new TEItemFrame();
				break;
			case 2:
				tileEntity = new TELogicSensor();
				break;
// however, it turns out that the variable **b** can be 255 (in my situation), so the **tileEntity** will never be assigned.
			}
// NullReferenceException
			tileEntity.type = b;
			tileEntity.ReadInner(reader, networkSend);
			return tileEntity;
		}
```
**I've noticed that you have done the WorldGen merging, so I thought that I would better submit an issue to tell you.**
I tried to solve this problem, however, i failed.
Thanks for all your hard working, and I am looking forward to the new TerrariaServerApi.

**Comment by tylerjwatson** [comment]
I can load an old map fine. Can you please attach the world file in question here so I have a 100% reproduce case?

**Comment by mistzzt** [comment]
I cannot upload the map file while github told me they don't support that file type, so I uploaded it to the [Google Drive Download](https://drive.google.com/file/d/0B6f1VXlwHWB9YVg4UDIxQk1mc1k/view?usp=sharing). 
1. I **can** load this map file in Terraria and TerrariaServer.
2. I tried to modify this map in Terraria Game and re-saved it, but it could not be loaded in TrServerApi.
3. I also tried backups, but i failed either..
![problem](https://cloud.githubusercontent.com/assets/12794966/15564378/2754daf0-2344-11e6-8abe-be75ea718245.png)
